how-to/how-to-design-database-schema.md

```diff
@@ -0,0 +1,304 @@
+- The data duplication that the embedding results in does not provide sufficient read performance advantages.
+- You want to represent more complex many-to-many relationships<br />In many-to-many cases, it is generally preferred to use separate content types. Consider Movies and Actors: each movie can feature multiple actors and each actor can star in multiple movies. Even though using embedded data in this case is technically possible, it would require updates to multiple records.
+- You want to model large hierarchical data sets.
+
+In Telerik Platform, you create relationships by storing only the "*id*" of a document from the related content type instead of the whole document. You then compose a single query to read data from the related content type and include a special expand clause with it. It is important to note that even though you are sending a single query, internally the server is executing two or more. This can negatively impact your app's performance, especially if you use nested expands. To prevent slow performance, Telerik Platform enforces restrictions to data expansion.
+
+#### Restrictions when Using Relations
+
+Telerik Platform poses a few restrictions when working with related content types which aim to guarantee the best performance in all situations. Make sure you take them into consideration when creating your schema design.
+
+The main restrictions are:
+
+- With the exception of "read single item" operations, you can only filter data coming from the content type that you are making the request to. The data from the related content type is returned in its entirety.<br \>Consider the following 2 scenarios:<br \>**Normalized data** You have a pair of content types: "Movie" and "Actor" and each movie record contains a relation to the "Actor" content type for each member of the cast. The restriction in this case will mean that you'll be able to filter for movies containing "operation" in their name but not for movies in which actors born 1970 are starring.<br \>**Denormalized data**
+If we used the embedded data model (all actor data was a part of the "Movie" content type) there would be no restrictions to filtering on any movie or actor field.
+- Result item count is restricted to 50 items.<br \>Because of this, you are advised to be as specific as possible when composing queries and let the server do the filtering instead of reading all data and filtering it on the device.
+- Go to the [Telerik Platform documentation](http://docs.telerik.com/platform/backend-services/javascript/data/relations/relations-limitations) to see the full list of restrictions.
+
+
+#### Schema Design in Action: Forum Posts Example
+
+Had enough theory? Let’s jump to something more akin to a real-life app. Assume we want to create the social app discussed above. It will allow users to create posts and each post will collect comments and likes from other users. How will its database schema look like in Telerik Platform?

```

Comment:
I feel that we may need to change the writing style from a blog post to a KB. Let's see what the others will say.